common/chat.ts

```diff
@@ -127,6 +132,12 @@ export interface ChannelInfo {
   id: SbChannelId
   /** The name of the chat channel. */
   name: string
+  /** The description of the chat channel. */
```

Comment:
If these don't exist in the model and you're not adding them here, you should at least file an issue to do that

Reply:
This made me think, should this information be available under `joinedChannelData` instead? In other words, should people be able to see banner images, badge images, and description of private channels they're not in?

Reply:
Yes they should be able to? I mean how are you planning to make any kind of useful channel directory if you treat all of that data as private to people that have joined the channel?

Reply:
I was thinking of excluding private channels from the search results of chat channels. My thinking was that you'd need an invite to join a private channel anyway, so wasn't sure what was the point of showing it in the search results.

Also, `joinedChannelData` is returned for people who have joined a private chat channel, so they should still be able to see this information.

Reply:
I don't really get what you're saying? You're saying you want to move `description` to a piece of data that is only returned once you have joined a channel, but we want `description` to be available in search results (and in the cards you are working on in this PR). So it makes no sense to move it there? I don't get what the private channel considerations have to do with this at all, other than that it needs to be optional as a result

Reply:
Yeah, I guess I haven't expressed myself all too clearly :d.

My thinking was that we should _not_ display `description` (and these other fields) for private channels (we can instead show a piece of content indicating the channel is private that you suggested above). This doesn't matter all too much for search results because I thought we were gonna exclude private channels from search results anyway. It only matters if someone mentions a private channel in chat where anyone can click on it and see its info card.

I guess I was confusingly thinking of `joinedChannelData` as "private" data, i.e. data that is returned for `private` channels as well, but only if the user has joined the channel. This would mean that the user who has joined the channel would see the full info in the channel card, and those who have not joined the channel would only see some limited information.

Reply:
No, the way that would be handled is that `description` would not be filtered out for users that were in a private channel. I don't see what this has to do with JoinedChannelData at all

Reply:
Well, I thought all "private" channel data, i.e. data that we don't want users to see unless they're joined in a channel, would go into the `JoinedChannelData` structure. I was unsure if `description` should be treated like that.

Reply:
Hmm, I think I finally got what you mean. And my last reply here is wrong, since we already have the `userCount` field which fits that description, but is not inside the `JoinedChannelData` structure. I assume you were saying `description` could be like that?

I think the problem lies in ambiguity of `JoinedChannelData` structure, and I think we could probably remove that structure altogether. I _think_ what I initially meant by `JoinedChannelData` structure is to represent data that is only returned when the user is subscribed to the channel. But this seems weird to separate data in structure based on APIs.

My new thinking is that we could have a flat `ChatInfo` type, where all the data that is not returned for private channels is simply optional, e.g.:
```ts
interface ChannelInfo {
  id: SbChannelId
  name: string
  private: boolean
  official: boolean
  description?: string
  topic?: string
  ownerId?: SbUserId
  bannerPath?: string
  badgePath?: string
  userCount?: number
}
```
So `GetChannelInfo(s)` would return either only required fields for private channels, or all the fields for non-private channels (and private channels that user is joined in).
And subscribing user to a channel would give them the full channel info of course (although weirdly enough without `userCount` atm, since we compute that inline in `GetChannelInfo` method).

I'm not sure if this is better since there's still some weirdness here (e.g. `ownerId` being `undefined` can mean the channel is private _or_ the channel is official and simply doesn't have an owner; although that field will never be used to determine that since we have separate flags for that).

Reply:
How is `JoinedChannelData` ambiguous? `JoinedChannelData` is `Data` that you only get for `Channel`s that you have `Joined`. The only way it is ambiguous is that you are trying to put stuff in it now that is *not* that (i.e. `description`, which is very much something you need for channels you have *not* joined).

Please just leave these things how they are and finish this PR instead of trying to endlessly re-architect these things (and either populate the `description` field properly or file an issue for it, like the original comment in this thread was talking about).